exercises/yacht/examples/success-standard/src/Yacht.hs

```diff
@@ -0,0 +1,69 @@
+module Yacht (yacht, Category(..)) where
+
+import Data.List
+
+data Category = Ones
+              | Twos
+              | Threes
+              | Fours
+              | Fives
+              | Sixes
+              | FullHouse
+              | FourOfAKind
+              | LittleStraight
+              | BigStraight
+              | Choice
+              | Yacht
+
+yacht :: Category -> [Int] -> Int
+yacht _ []   = error "You lose!"
+yacht Ones   dice = simpleScore 1 dice
+yacht Twos   dice = simpleScore 2 dice
+yacht Threes dice = simpleScore 3 dice
+yacht Fours  dice = simpleScore 4 dice
+yacht Fives  dice = simpleScore 5 dice
+yacht Sixes  dice = simpleScore 6 dice
+
+yacht FullHouse dice
+  | s == 2 && length g == 2 && (length . head') gs == 3 = sum dice
+  | otherwise = 0
+  where
+    s = (length . gsortBySize) dice
+    (g:gs) = gsortBySize dice
```

Comment:
What if `dice` is empty?

Reply:
Ah yes, I had been using another function I built that reduced all duplicates, but I figured I didn't need both.  

Will the composition with `length` disrupt memorization?

So are we both thinking save it, then get the length and then split it into it's parts?

Also, for dice dice is empty (I ignored it since it is untested by the problem-specifications tests): it would be enough to have:
``` Haskell
yacht _ [] = error "useful error message here"
```
Before all of the other patterns for `yacht`?

Reply:
I forgot I already used the above pattern with `"You lose!"` as the error message.  I figured that message might need to change (although it is untested, so I don't think it is that important).

Does that not handle cases where dice is empty?

Also, do you have a preference for the error message?

Reply:
Just in case we're not in agreement here: I suggest that the function returns 0 if the input is invalid, rather than try to narrow down the input type to exactly 5d6, and rather than make the function return `Maybe Score` or `Either String Score`.